kvbc/include/categorization/kv_blockchain.h

```diff
@@ -24,88 +24,75 @@ namespace concord::kvbc::categorization {
 
 class KeyValueBlockchain {
  public:
-  KeyValueBlockchain()
-      : native_client_(concord::storage::rocksdb::NativeClient::newClient(
-            "/tmp", false, concord::storage::rocksdb::NativeClient::DefaultOptions{})) {}
   KeyValueBlockchain(const std::shared_ptr<concord::storage::rocksdb::NativeClient>& native_client)
-      : native_client_(native_client) {}
-  // 1) Defines a new block
-  // 2) calls per cateogry with its updates
-  // 3) inserts the updates KV to the DB updates set per column family
-  // 4) add the category block data into the new block
-  BlockId addBlock(Updates&& updates) {
-    // Use new client batch and column families
-    auto write_batch = native_client_->getBatch();
-    Block new_block{getLastReachableBlockId() + 1};
-    // auto parentBlockDigestFuture = computeParentBlockDigest(new_block.ID( ));
-    // Per category updates
-    for (auto&& [category_id, update] : updates.category_updates_) {
-      // https://stackoverflow.com/questions/46114214/lambda-implicit-capture-fails-with-variable-declared-from-structured-binding
-      std::visit(
-          [&new_block, category_id = category_id, &write_batch, this](auto& update) {
-            auto block_updates = handleCategoryUpdates(new_block.id(), category_id, std::move(update), write_batch);
-            new_block.add(category_id, std::move(block_updates));
-          },
-          update);
-    }
-    if (updates.shared_update_.has_value()) {
-      auto block_updates =
-          handleCategoryUpdates(new_block.id(), std::move(updates.shared_update_.value()), write_batch);
-      new_block.add(std::move(block_updates));
-    }
-    // newBlock.parentDigest = parentBlockDigestFuture.get();
-    // // E.L blocks in new column Family
-    write_batch.put(detail::BLOCKS_CF, Block::generateKey(new_block.id()), Block::serialize(new_block));
-    native_client_->write(std::move(write_batch));
-    return lastReachableBlockId_ = new_block.id();
-  }
+      : native_client_{native_client},
+        block_chain_{*native_client_.get()},
+        state_transfer_block_chain_{*native_client_.get()} {}
+
+  /////////////////////// Add Block ///////////////////////
+
+  BlockId addBlock(Updates&& updates);
+
+  /////////////////////// Delete block ///////////////////////
+
+  bool deleteBlock(const BlockId& blockId);
```

Comment:
I think since we don't need to support the IDbAdapter interface anymore, we might split to `deleteLastReachable()` and `deleteGenesis()`.

Reply:
As discussed offline, the deleteBlock method has state transfer blocks handling as well.
We'll define an appropriate interface for the deletion.